### DISCUSSION TITLE
How to make Pydantic's BaseModel accept floats without numbers after & before period, that numpy does, but Pydantic's BaseModel raises `ReuestValidationError` on

### DISCUSSION THREAD

**Discussion by naveenmarthala** [pydantic/pydantic#3722] [Question]
I have a fastapi app, which takes in "application/json" among other input types and returns "appliction/json". I have a input-validator written sub-classing pydantic's `BaseModel` and I would like to make it accept some cases of inputs which I do receive and not generate `fastapi.exceptions.RequestValidationError` on them.

some accepted inputs (`List[List[float]]`):
1. `{"input_json":[[1.0, 2.0, 3.0, 4.0]]}`
2. `{"input_json":[[1.0, 2.0, 3.0, 4.0], [5.0, 6.0, 7.0, 8.0]]}`

I would like to make it accept following input cases:

1. `{"input_json":[[1.0, 2.0, 3.0, 4.], [5.0, 6.0, .0, 8.0]]}` (no numbers after and before decimal point. numpy arrays accept such inputs by populating 0 before and after decimal points, but pydantic throws validation error.)

Here's how my validaiton class is written:
```
from pydantic import BaseModel, validator

class InputResponse(BaseModel):
    input_json: List[List[float]]

    @validator("input_json")
    def check_dimensionality(cls, list_of_lists):
        for nested_list in list_of_lists:
            if len(nested_list) != 4:
                raise ValueError(incomplete_msg)
        return list_of_lists
```

what changes do I make to be able to accept both of those input cases?

pip versions on python-3.8.12 on debian-11:
```
fastapi==0.71.0
gunicorn==20.1.0
h11==0.12.0
httptools==0.3.0
numpy==1.22.0
pydantic==1.9.0
python-dotenv==0.19.2
PyYAML==6.0
sniffio==1.2.0
starlette==0.17.1
threadpoolctl==3.0.0
typing_extensions==4.0.1
uvicorn==0.17.0
uvloop==0.16.0
```

**Comment by NowanIlfideme** [comment]
Use s `@validator(FIELDNAME, pre=True) ` to get the **raw** value of your input. I'm not sure whether it will work in this case, but I believe FastAPI uses Pydantic to parse the json directly, so the raw value might be a string, and you can do whatever.

**Reply by naveenmarthala** [reply]
@NowanIlfideme, no `pre=True` is not working in my case.
